### PR TITLE
LPS-160835 [PRM] Create fields startDate and endDate in object mdfrequests

### PR DESCRIPTION
**Pull request by joaosvassis**

### CODE REVIEW

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/routes/MDFRequestForm/steps/Review/utils/GetTotalBudget.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import {useMemo} from 'react';
+
+import MDFRequest from '../../../../../common/interfaces/mdfRequest';
+import MDFRequestActivity from '../../../../../common/interfaces/mdfRequestActivity';
+
+const GetTotalBudget = (values: MDFRequest) => {
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/routes/MDFRequestForm/steps/Review/utils/GetTotalBudget.tsx:17]:
```suggestion
export default function getTotalBudget = (mdfRequest: MDFRequest) => {
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/routes/MDFRequestForm/steps/Review/utils/GetTotalBudget.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+					return previousValue + sumBudgets;
+				},
+				0
+			),
+		[values.activities]
+	);
+
+	return totalBudget;
+};
+export default GetTotalBudget;
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/routes/MDFRequestForm/steps/Review/utils/GetTotalBudget.tsx:37]:
```suggestion
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/routes/MDFRequestForm/steps/Review/utils/GetTotalBudget.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+					);
+
+					return previousValue + sumBudgets;
+				},
+				0
+			),
+		[values.activities]
+	);
+
+	return totalBudget;
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/routes/MDFRequestForm/steps/Review/utils/GetTotalBudget.tsx:35]:
```suggestion
	return useMemo(
		() =>
			values.activities.reduce(
				(previousValue: number, currentValue: MDFRequestActivity) => {
					const sumBudgets = currentValue.budgets.reduce<number>(
						(previousValue, currentValue) =>
							previousValue + +currentValue.cost,
						0
					);

					return previousValue + sumBudgets;
				},
				0
			),
		[values.activities]
	);
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/common/services/liferay/object/mdf-requests/createMDFRequest.ts [hunk 4]
```diff
@@ -10,21 +10,69 @@
...
+import MDFRequestActivity from '../../../../interfaces/mdfRequestActivity';
 import {LiferayAPIs} from '../../common/enums/apis';
 import liferayFetcher from '../../common/utils/fetcher';
 
+interface DateActivities {
+	endDates: Date[];
+	startDates: Date[];
+}
+
+const getDateActivities = (mdfRequest: MDFRequest) => {
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/common/services/liferay/object/mdf-requests/createMDFRequest.ts:24]: @joaosvassis, should be good add this function in utils directory with the interface too.

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/common/hooks/useTotalBudget.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import {useMemo} from 'react';
+
+import MDFRequest from '../interfaces/mdfRequest';
+import MDFRequestActivity from '../interfaces/mdfRequestActivity';
+
+const useTotalBudget = (mdfRequest: MDFRequest) => {
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/common/hooks/useTotalBudget.tsx:17]:
```suggestion
export default function useTotalBudget(mdfRequest: MDFRequest) {
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/common/hooks/useTotalBudget.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+					);
+
+					return previousValue + sumBudgets;
+				},
+				0
+			),
+		[mdfRequest.activities]
+	);
+};
+export default useTotalBudget;
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/common/hooks/useTotalBudget.tsx:35]:
```suggestion
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/common/services/liferay/object/mdf-requests/createMDFRequest.ts [hunk 7]
```diff
@@ -11,20 +11,27 @@
...
 import MDFRequest from '../../../../interfaces/mdfRequest';
+import getActivitiesData from '../../../../utils/getActivitiesData';
 import {LiferayAPIs} from '../../common/enums/apis';
 import liferayFetcher from '../../common/utils/fetcher';
 
+interface DateActivities {
+	endDates: Date[];
+	startDates: Date[];
+}
+
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/common/services/liferay/object/mdf-requests/createMDFRequest.ts:22]:
```suggestion
```

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

**Comment by DiogoNo** [comment]
Just starting reviewing :)

**Pull request closed without merging**